src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -996,11 +996,12 @@ JRT_ENTRY_NO_ASYNC(void, SharedRuntime::register_finalizer(JavaThread* current,
 JRT_END
 
 jlong SharedRuntime::get_java_tid(Thread* thread) {
```

Comment:
Additional RFE: can we not declare this to take JavaThread and so avoid the checks and casts below?

Reply:
I can research that and file a follow up RFE if necessary.

Reply:
New RFE:
[JDK-8289004](https://bugs.openjdk.org/browse/JDK-8289004) investigate if SharedRuntime::get_java_tid parameter should be a JavaThread*